fgpyo/sequence.py

```diff
@@ -73,7 +88,11 @@ def reverse_complement(bases: str) -> str:
     Returns:
         the reverse complement of the provided base string
     """
-    return "".join([_COMPLEMENTS[b] for b in bases[::-1]])
+    rev_comp = bases.translate(_COMPLEMENTS_TABLE)[::-1]
+    if len(rev_comp) != len(bases):
+        # There were invalid characters that weren't translated. Find one and raise KeyError.
+        raise KeyError(next(base for base in bases if base not in _COMPLEMENTS))
```

Comment:
It would be nice to list out all the failed bases, not just the first one found.

Something like:

```suggestion
        raise ValueError("Invalid bases found: " + ", ".join(sorted(set(base for base in bases if base not in _COMPLEMENTS))))
```

But I do understand you tried to keep backwards compatibility with the `KeyError`.

Reply:
I kept `KeyError` for backward compatibility, but gave a better description. Basically yours but I used a `set` instead of a `list`, so that you get all the _distinct_ errors but not potentially 1 Mbp of "there's no Z in the genome".